src/bootstrap/src/core/config/config.rs

```diff
@@ -2028,13 +2030,19 @@ impl Config {
                 compression_formats,
                 compression_profile,
                 include_mingw_linker,
+                vendor,
             } = dist;
             config.dist_sign_folder = sign_folder.map(PathBuf::from);
             config.dist_upload_addr = upload_addr;
             config.dist_compression_formats = compression_formats;
             set(&mut config.dist_compression_profile, compression_profile);
             set(&mut config.rust_dist_src, src_tarball);
-            set(&mut config.dist_include_mingw_linker, include_mingw_linker)
+            set(&mut config.dist_include_mingw_linker, include_mingw_linker);
+            config.dist_vendor = vendor.unwrap_or_else(|| {
+                // If we're building from git or tarball sources, enable it by default.
```

Comment:
What's the alternative option to these two? Someone clones git and then deletes .git?

Reply:
> Someone clones git and then deletes .git?

That might be a bit awkward, but it’s still one possibility.

Another case that fits this scenario is using this download button:

<img width="346" alt="Ekran görüntüsü 2024-09-14 182025" src="https://github.com/user-attachments/assets/de2648d5-d14e-4194-9cc2-90a058d275dc">

Reply:
> Another case that fits this scenario is using this download button:

That produces a broken source archive as all git submodules missing. You can't even build the standard library from that archive as both stdarch (`core::arch`) and backtrace (`std::backtrace`) are submodules.